crates/hyperdrive-math/src/short/open.rs

```diff
@@ -44,117 +70,197 @@ impl State {
-            .mul_up(self.calculate_short_principal(bond_amount)?);
+        // Calculate the effect that opening the short will have on the pool's
+        // share reserves.
+        let share_reserves_delta = self.calculate_short_principal(bond_amount)?;
+
+        // NOTE: Round up to make the check stricter.
+        //
         // If the base proceeds of selling the bonds is greater than the bond
         // amount, then the trade occurred in the negative interest domain.
         // We revert in these pathological cases.
-        if share_reserves_delta_in_base > bond_amount {
+        if share_reserves_delta.mul_up(self.vault_share_price()) > bond_amount {
             return Err(eyre!(
                 "InsufficientLiquidity: Negative Interest.
-                expected bond_amount={} <= share_reserves_delta_in_base={}",
+                expected bond_amount={} <= share_reserves_delta_in_shares={}",
                 bond_amount,
-                share_reserves_delta_in_base
+                share_reserves_delta
             ));
         }
 
-        // NOTE: The order of additions and subtractions is important to avoid underflows.
-        let spot_price = self.calculate_spot_price()?;
-        Ok(
-            bond_amount.mul_div_down(self.vault_share_price(), open_vault_share_price)
-                + self.flat_fee() * bond_amount
-                + self.curve_fee() * (fixed!(1e18) - spot_price) * bond_amount
-                - share_reserves_delta_in_base,
-        )
+        // NOTE: Round up to overestimate the base deposit.
+        //
+        // The trader will need to deposit capital to pay for the fixed rate,
+        // the fees, and any back-paid interest that will be received back upon
+        // closing the trade.
+        let curve_fee_shares = self
+            .open_short_curve_fee(bond_amount)?
+            .div_up(self.vault_share_price());
+
+        if share_reserves_delta < curve_fee_shares {
+            return Err(eyre!(format!(
+                "The transaction curve fee = {}, computed with coefficient = {},
+                is too high. It must be less than share reserves delta = {}",
+                curve_fee_shares,
+                self.curve_fee(),
+                share_reserves_delta
+            )));
+        }
+
+        // If negative interest has accrued during the current checkpoint, we
+        // set the close vault share price to equal the open vault share price.
+        // This ensures that shorts don't benefit from negative interest that
+        // accrued during the current checkpoint.
+        let close_vault_share_price = open_vault_share_price.max(self.vault_share_price());
+
+        // Now we can calculate adjusted proceeds account for the backdated
+        // vault price:
+        //
+        // $$
+        // \text{base_proceeds} = (
+        //    \frac{c1 \cdot \Delta y}{c0 \cdot c}
+        //    + \frac{\Delta y \cdot \phi_f}{c} - \Delta z
+        // ) \cdot c
+        // $$
```

Comment:
Why are we using latex for these comments? It isn't going to get rendered, so it feels like it's just making it harder to read.

Reply:
I'm going to leave this for now. I added a comment about it here https://github.com/delvtech/hyperdrive-rs/issues/27